utils/httpx/access.go

```diff
@@ -0,0 +1,37 @@
+package httpx
+
+import (
+	"net"
+	"net/http"
+	"strings"
+)
+
+// AccessConfig configures what can be accessed
+type AccessConfig struct {
+	DisallowedIPs []net.IP
+}
+
+// NewAccessConfig creates a new access config
+func NewAccessConfig(disallowedIPs []net.IP) *AccessConfig {
+	return &AccessConfig{DisallowedIPs: disallowedIPs}
+}
+
+// Allow determines whether the given request should be allowed
+func (c *AccessConfig) Allow(request *http.Request) (bool, error) {
+	host := strings.ToLower(request.URL.Hostname())
+
+	addrs, err := net.LookupIP(host)
```

Comment:
I think we probably want to use https://golang.org/pkg/net/#Resolver.LookupIPAddr because otherwise there's no timeout on this and we can just hang for any numbers of reasons.